spring-messaging/src/main/java/org/springframework/messaging/handler/invocation/reactive/InvocableHandlerMethod.java

```diff
@@ -129,11 +131,13 @@ public void setReactiveAdapterRegistry(ReactiveAdapterRegistry registry) {
 			Object value;
 			try {
 				Method method = getBridgedMethod();
-				ReflectionUtils.makeAccessible(method);
 				if (KotlinDetector.isKotlinReflectPresent() && KotlinDetector.isKotlinType(method.getDeclaringClass())) {
-					value = CoroutinesUtils.invokeSuspendingFunction(method, getBean(), args);
+					KFunction<?> function = KotlinReflectionUtils.methodToFunction(method);
+					KotlinReflectionUtils.makeFunctionAccessible(function);
+					value = CoroutinesUtils.invokeSuspendingFunction(function, getBean(), args);
 				}
 				else {
+					ReflectionUtils.makeAccessible(method);
```

Comment:
The code for this has moved to the constructor of `HandlerMethod` so this needs to be removed from here.